lib/rouge/lexers/nial.rb

```diff
@@ -0,0 +1,166 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+    module Lexers
+        class Nial < RegexLexer
+            title 'Nial'
+            desc 'The Nial programming language (nial-array-language.org)'
+            tag 'nial'
+            filenames '*.ndf', '*.nlg'
+            def self.keywords
+                @keywords ||= Set.new ["is", "gets", "op", "tr", ";",
+                                       "if", "then", "elseif", "else",
+                                       "endif", "case", "from", "endcase",
+                                       "begin", "end", "for", "with",
+                                       "endfor", "while", "do", "endwhile",
+                                       "repeat", "until", "endrepeat"]
+            end
+
+            def self.operators
+                @operators||= Set.new [".",  "!", "#", "+", "*", "-", "<<",
+                                 "/", "<", ">>", "<=", ">", "=", ">=", "@", "|", "~="]
+            end
+
+            def self.punctuation
```

Comment:
**suggestion:* Could we rename this to a plural name i.e, `punctuations`?

Reply:
renamed.